plugin/save_command.py

```diff
@@ -90,6 +90,9 @@ def run(self, edit: sublime.Edit) -> None:
         else:
             self._trigger_native_save()
 
+    def is_enabled(self, event: Optional[dict] = None, point: Optional[int] = None) -> bool:
+        return True
```

Comment:
I don’t understand how this solves the linked issue. Can you add some comments in code to explain why this is needed?

Reply:
Looks like after detaching a document, the `on_query_context` in:
https://github.com/sublimelsp/LSP/blob/0153663813015100013bbe44e9903d81b0f1b0a5/plugin/documents.py#L408-L408
will return `True` but `LspTextCommand.is_enabled`:
https://github.com/sublimelsp/LSP/blob/0153663813015100013bbe44e9903d81b0f1b0a5/plugin/core/registry.py#L68-L68
return `False`.

So it looks like those places check for sessions in a different one and one way is affected by the detach bug and the other isn't.

Reply:
`DocumentSyncListener.on_query_context` returns `True` if a view is dragged to a new window, while `LspTextCommand.is_enabled` returns `False`. I don't know why this divergence happens, but there must be something fundamentally wrong with how applicable views are managed in a session or so. Probably due to the mentioned ST core issue that `on_post_move_async` isn't triggered. Anyway, `on_query_context` will cause the keybinding to be active, while `is_enabled` prevents the `run` method of `lsp_save` to be executed. This turns <kbd>Ctrl</kbd>+<kbd>S</kbd> into "noop". By enforcing `is_enabled` to be `True` is it at least ensured that the command is run, and the file can be saved. The bug affected users even if they don't use any code actions on save at all.

It should be pretty fine to have `is_enabled` always be `True`, because it is only used for that keybinding and not in code via `run_command("lsp_save")`. In general, it should not make much sense to do the same check twice whether the keybinding is applicable or not, both in `on_query_context` and in `is_enabled`. Eventually a proper fix would ensure that both will return the same value for `lsp_save`, so that the code actions on save also work as expected (not sure if they already work or not - I just wanted to fix the linked issue here).

I don't really know if the actual bug is in `DocumentSyncListener.on_query_context` or in `LspTextCommand.is_enabled` or somewhere else.

Reply:
By the way, this bug affects other things too. For example DocumentHighlights still seem to work fine, while most other things like hover won't work anymore with such a view.